test/tests/pbs_admin_suspend.py

```diff
@@ -405,3 +413,416 @@ def test_degraded_resv(self):
+        j = Job(TEST_USER)
+        jid1 = self.server.submit(j)
+        jid2 = self.server.submit(j)
+        jid3 = self.server.submit(j)
+
+        # verify that all are running
+        self.server.expect(JOB, {'job_state=R': 3, 'substate=42': 3})
+
+        # admin-suspend 2 of them
+        self.server.sigjob(jid2, 'admin-suspend', runas=ROOT_USER)
+        self.server.sigjob(jid3, 'admin-suspend', runas=ROOT_USER)
+
+        # node state is in maintenance
+        self.server.expect(NODE, {'state': 'maintenance'}, id='vn[0]')
+
+        # list maintenance_jobs as root
+        self.server.expect(NODE, {'maintenance_jobs': jid2 + "," + jid3},
+                           runas=ROOT_USER)
+
+        # list maintenance jobs as user
+        self.server.expect(NODE, {'maintenance_jobs': jid2 + "," + jid3},
+                           runas=TEST_USER)
+
+        # set an operator
+        self.server.manager(MGR_CMD_SET, SERVER, {'operators': 'pbsoper@*'})
+
+        # List all jobs in maintenance mode as operator
+        self.server.expect(
+            NODE, {'maintenance_jobs': jid2 + "," + jid3}, runas='pbsoper')
+
+        # set maintenance_jobs as root
+        try:
+            self.server.manager(MGR_CMD_SET, NODE,
+                                {'maintenance_jobs': jid1}, id='vn[0]',
+                                runas=ROOT_USER)
+        except PbsManagerError, e:
+            self.assertFalse(e.rv)
+            msg = "Cannot set attribute, read only" +\
+                  " or insufficient permission  maintenance_jobs"
+            self.assertTrue(msg in e.msg[0])
+
+        # Set maintenance_jobs as operator
+        try:
+            self.server.manager(MGR_CMD_SET, NODE,
+                                {'maintenance_jobs': jid1}, id='vn[0]',
+                                runas='pbsoper')
+        except PbsManagerError, e:
+            self.assertFalse(e.rv)
+            msg = "Cannot set attribute, read only" +\
+                  " or insufficient permission  maintenance_jobs"
+            self.assertTrue(msg in e.msg[0])
+
+        # Set maintenance_jobs as user
+        try:
+            self.server.manager(MGR_CMD_SET, NODE,
+                                {'maintenance_jobs': jid1}, id='vn[0]',
+                                runas=TEST_USER)
+        except PbsManagerError, e:
+            self.assertFalse(e.rv)
+            self.assertTrue("Unauthorized Request" in e.msg[0])
+
+    def test_list_jobs_2(self):
+        """
+        Test to list maintenance_jobs when no job is admin-suspended
+        """
+
+        # Submit a few jobs
+        j = Job(TEST_USER)
+        jid1 = self.server.submit(j)
+        jid2 = self.server.submit(j)
+        jid3 = self.server.submit(j)
+
+        # verify that all are running
+        self.server.expect(JOB, {'job_state=R': 3, 'substate=42': 3})
+
+        # list maintenance_jobs. It should be empty
+        try:
+            self.server.expect(
```

Comment:
I've always tested for unset attributes with using the parameter op=UNSET.  I've not seen someone test for "" before.